src/cards/cover-card/cover-card.ts

```diff
@@ -205,6 +204,34 @@ export class CoverCard extends MushroomBaseElement implements LovelaceCard {
         `;
     }
 
+    private renderIcon(entity: CoverEntity, icon: string, available: boolean): TemplateResult {
+        const iconStyle = {
+            "--icon-color": "rgb(var(--rgb-state-cover))",
+            "--shape-color": "rgba(var(--rgb-state-cover), 0.2)",
+        };
+
+        const currentState = this.getStyleState(entity);
+        if (currentState) {
+            iconStyle["--icon-color"] = `rgb(var(--rgb-state-cover-${currentState}))`;
+            iconStyle["--shape-color"] = `rgba(var(--rgb-state-cover-${currentState}), 0.2)`;
+        }
+
+        return html`
+            <mushroom-shape-icon
+                slot="icon"
+                .disabled=${!available}
+                .icon=${icon}
+                style=${styleMap(iconStyle)}
+            ></mushroom-shape-icon>
+        `;
+    }
+
+    private getStyleState(entity: CoverEntity): string | null {
+        if (isFullyOpen(entity) || isOpening(entity)) return "open"
+        if (isFullyClosed(entity) || isClosing(entity)) return "closed"
+        return null;
+    }
```

Comment:
```suggestion
    private getStyleState(entity: CoverEntity): string | null {
        if (isOpen(entity) || isOpening(entity)) return "open"
        if (isClosed(entity) || isClosing(entity)) return "closed"
        return null;
    }
```
I think the rules must be these ones.

Reply:
`isOpen` and `isClosed` methods does not exist in `cover.py`, I simplified things now by just checking the states.